vars/deploy.groovy

```diff
@@ -0,0 +1,149 @@
+            .mixin(parameters, PARAMETER_KEYS)
+            .use()
+
+        utils.pushToSWA([
+            step: STEP_NAME,
+            stepParamKey1: 'stage',
+            stepParam1: stageName,
+            stepParamKey2: 'enableZeroDowntimeDeployment',
+            stepParam2: enableZeroDowntimeDeployment
+        ], config)
+
+        def index = 1
+        def deployments = [:]
+        def deploymentType
+        def deployTool
+
+        if (config.cfTargets) {
+
+            deploymentType = DeploymentType.selectFor(CloudPlatform.CLOUD_FOUNDRY, enableZeroDowntimeDeployment).toString()
+            deployTool = script.commonPipelineEnvironment.configuration.isMta ? 'mtaDeployPlugin' : 'cf_native'
+
+            for (int i = 0; i < config.cfTargets.size(); i++) {
+
+                def target = config.cfTargets[i]
+
+                Closure deployment = {
+
+                    utils.unstash(stageName)
+
+                    cloudFoundryDeploy(
+                        script: script,
+                        juStabUtils: utils,
+                        jenkinsUtilsStub: jenkinsUtils,
+                        deployType: deploymentType,
+                        cloudFoundry: target,
+                        mtaPath: script.commonPipelineEnvironment.mtarFilePath,
+                        deployTool: deployTool
+                    )
+
+                    utils.stash(stageName)
+                }
+                runDeployment(deployments, deployment, index, script, stageName)
+                index++
+            }
+            utils.runClosures(deployments)
+        }
+
+        if (config.neoTargets) {
+
+            deploymentType = DeploymentType.selectFor(CloudPlatform.NEO, enableZeroDowntimeDeployment)
+
+            def source
+            if (config.source) {
+                source = config.source
+            } else {
+                def pom = readMavenPom file: 'application/pom.xml'
+                source = "application/target/${pom.getArtifactId()}.${pom.getPackaging()}"
+            }
+
+            for (int i = 0; i < config.neoTargets.size(); i++) {
+
+                def target = config.neoTargets[i]
+
+                Closure deployment = {
+
+                    utils.unstash(stageName)
+
+                    neoDeploy (
+                        script: script,
+                        warAction: deploymentType.toString(),
+                        source: source,
+                        neo: target
+                    )
+
+                    utils.stash(stageName)
+                }
+                runDeployment(deployments, deployment, index, script, stageName)
+                index++
+            }
+            utils.runClosures(deployments)
+        }
+
+        if (!config.cfTargets && !config.neoTargets) {
+            error "Deployment skipped because no targets defined!"
+        }
+    }
+
+}
+
+void runDeployment(deployments, deployment, index, script, stageName) {
```

Comment:
I was confused about the name of the method as it just modifies the deployments map.

IMO, I would vote for not modifying the map but return something which is stored in the map outside of the scope of this method